File: Data Structures & Algorithms/sliding-window-maximum/submission-1.py
class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:
        '''
        brute force
        '''
        res = []
        for i in range(0, len(nums) - k + 1):
            temp_max = float('-infinity')
            for j in range(i, i + k):
                temp_max = max(temp_max, nums[j])
            res.append(temp_max)
        return res

File: Data Structures & Algorithms/house-robber-ii/submission-1.cpp
class Solution
{
public:
    int rob(vector<int> &nums)
    {
        int size = nums.size();
        if (size == 1)
        {
            return nums[0];
        }
        int minus_first[size - 1];
        int minus_last[size - 1];

        /* minus_first house */
        minus_first[0] = nums[1];
        minus_first[1] = max(nums[1], nums[2]);
        for (int i = 2; i < size - 1; i++)
        {
            minus_first[i] = max(minus_first[i - 1], minus_first[i - 2] + nums[i + 1]);
        }

        /* minus_last house */
        minus_last[0] = nums[0];
        minus_last[1] = max(nums[0], nums[1]);
        for (int i = 2; i < size - 1; i++)
        {
            minus_last[i] = max(minus_last[i - 1], minus_last[i - 2] + nums[i]);
        }

        return max(minus_last[size - 2], minus_first[size - 2]);
    }
};